pkgs/development/compilers/spirv-llvm-translator/default.nix

```diff
@@ -4,29 +4,42 @@
 , pkg-config
 , lit
 , llvm_11
+, spirv-headers
+, spirv-tools
 }:
 
 stdenv.mkDerivation rec {
   pname = "SPIRV-LLVM-Translator";
-  version = "unstable-2021-06-13";
+  version = "unstable-2022-05-04";
 
   src = fetchFromGitHub {
     owner = "KhronosGroup";
     repo = "SPIRV-LLVM-Translator";
-    rev = "c67e6f26a7285aa753598ef792593ac4a545adf9";
-    sha256 = "sha256-1s3lVNTQDl+pUvbzSMsp3cOUSm6I4DzqJxnLMeeE3F4=";
+    rev = "99420daab98998a7e36858befac9c5ed109d4920";
+    sha256 = "sha256-/vUyL6Wh8hykoGz1QmT1F7lfGDEmG4U3iqmqrJxizOg=";
   };
 
   nativeBuildInputs = [ pkg-config cmake llvm_11.dev ];
 
-  buildInputs = [ llvm_11 ];
+  buildInputs = [ spirv-headers spirv-tools llvm_11 ];
 
   checkInputs = [ lit ];
 
+  makeFlags = [ "llvm-spirv" ];
+
   cmakeFlags = [
     "-DLLVM_INCLUDE_TESTS=ON"
+    "-DLLVM_DIR=${llvm_11.dev}"
+    "-DBUILD_SHARED_LIBS=YES"
+    "-DLLVM_SPIRV_BUILD_EXTERNAL=YES"
   ];
 
+  prePatch = ''
+    substituteInPlace ./test/CMakeLists.txt \
+      --replace 'SPIRV-Tools' 'SPIRV-Tools-shared'
+  '';
```

Comment:
Removed in a follow-up

Reply:
This is copy&paste from https://github.com/calbrecht/f4s-fixups/blob/main/flake.nix#L126, i had the `lit` tool fixed and tests running until i noticed `clinfo` was segfaulting. Then i discovered that `spirv-llvm-translator` needs to be build with `-DBUILD_SHARED_LIBS=YES` and afterwards the tests were failing again because `lit` was unable to find the `libLLVMSPIRVLib.so.11` so i gave up on that.